File: examples/saxpy-parallel.f90
program saxpy
  integer, parameter :: N=1024
  real, dimension(N) :: x, y
  integer            :: i

  !$acc parallel loop
  do i=1,N
    y(i) = 0
    x(i) = i
  enddo

  !$acc parallel loop
  do i=1,N
    y(i) = 2.0 * x(i) + y(i)
  enddo

  print *, y(1), y(N)

end program saxpy

File: examples/saxpy-parallel.c
#include <stdio.h>
#define N 1024

int main(int argc, char **argv)
{
  float x[N], y[N];
  int i;

#pragma acc parallel loop
  for (i=0; i<N; i++)
  {
    y[i] = 0.0f;
    x[i] = (float)(i+1);
  }

#pragma acc parallel loop
  for (i=0; i<N; i++)
  {
    y[i] = 2.0f * x[i] + y[i];
  }

  printf("%f %f\n",y[0],y[N-1]);

  return 0;
}
